api/query/cache/index/filter.go

```diff
@@ -236,7 +159,7 @@ func (tnp SubtestNamePattern) Filter(t TestID) bool {
 }
 
 // Filter interprets a TestPath as a filter function over TestIDs.
-func (tp *TestPath) Filter(t TestID) bool {
```

Comment:
Can you explain why the various `*` and `&` changes in this file were first made (if you know), and if we need to or should revert them?

Reply:
The reverted change introduced the first mutation on a bound query (see `fcq.searchResults`), which required the receiver type to be a pointer, so `newFilter` had to return `&FileContentsQuery` to match the `filter` interface, but returning a pointer only in some cases can be confusing and error-prone in the future, so `newFilter` was changed to always return struct pointers. 

I don't have strong preference over either (these are tiny structs) but I'd prefer a clean revert here.

Reply:
Ok, revert sounds good to me in that case. Thanks!